src/Voron/Data/Tables/Table.cs

```diff
@@ -331,6 +331,12 @@ public bool IsOwned(long id)
 
         public void Delete(long id)
         {
+            if (IsOwned(id) == false)
```

Comment:
I'm worried that this will hit the performance, because this check will be done too often.

Reply:
I'm not sure, eventually we just compare hash here... we will load the page either way.
But maybe I'm wrong..

Reply:
We can usually assume that this is loaded to memory. Also, I don't think we have perf critical deletes usually.

Reply:
Very likely it's already loaded. Even if not we are about to do it anyway since we have `DirectRead()` below. We talked about it and IMO it's fine, although would be good to have @redknightlois input on this as well. In some other places we have the validation with the usage of `IsOwned()` only in Debug but here would be good to assert it in runtime to prevent strange errors.

Reply:
Original: 
```
|                           Method | KeyLength | GenerationDeletionProbability | GenerationTableSize | GenerationBatchSize | RandomSeed | NumberOfTransactions | NumberOfRecordsPerTransaction |     Mean |   Error |  StdDev |   Median |
|--------------------------------- |---------- |------------------------------ |-------------------- |-------------------- |----------- |--------------------- |------------------------------ |---------:|--------:|--------:|---------:|
|       DeleteRandomOneTransaction |       100 |                           0.1 |              500000 |               50000 |        357 |                 1000 |                          1000 | 164.4 ms | 1.55 ms | 1.45 ms | 164.2 ms |
| DeleteRandomMultipleTransactions |       100 |                           0.1 |              500000 |               50000 |        357 |                 1000 |                          1000 | 181.2 ms | 3.61 ms | 8.29 ms | 178.2 ms |
```
With my change:
```
|                           Method | KeyLength | GenerationDeletionProbability | GenerationTableSize | GenerationBatchSize | RandomSeed | NumberOfTransactions | NumberOfRecordsPerTransaction |     Mean |   Error |  StdDev |
|--------------------------------- |---------- |------------------------------ |-------------------- |-------------------- |----------- |--------------------- |------------------------------ |---------:|--------:|--------:|
|       DeleteRandomOneTransaction |       100 |                           0.1 |              500000 |               50000 |        357 |                 1000 |                          1000 | 165.9 ms | 2.73 ms | 2.42 ms |
| DeleteRandomMultipleTransactions |       100 |                           0.1 |              500000 |               50000 |        357 |                 1000 |                          1000 | 176.8 ms | 1.93 ms | 1.81 ms |
```
The benchmark: https://gist.github.com/karmeli87/ffd230d65358e59bd6090d382ac14a19

Reply:
Very well within the margin or error. Go with the change.